core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -234,6 +290,57 @@ private static void writeAddSortOrder(MetadataUpdate.AddSortOrder update, JsonGe
     SortOrderParser.toJson(update.sortOrder(), gen);
   }
 
+  private static void writeSetDefaultSortOrder(MetadataUpdate.SetDefaultSortOrder update, JsonGenerator gen)
+      throws IOException {
+    gen.writeNumberField(SORT_ORDER_ID, update.sortOrderId());
+  }
+
+  private static void writeAddSnapshot(MetadataUpdate.AddSnapshot update, JsonGenerator gen) throws IOException {
+    gen.writeFieldName(SNAPSHOT);
+    SnapshotParser.toJson(update.snapshot(), gen);
+  }
+
+  // TODO - Reconcile the spec's set-based removal with the current class implementation that only handles one value.
+  private static void writeRemoveSnapshots(MetadataUpdate.RemoveSnapshot update, JsonGenerator gen) throws IOException {
```

Comment:
Same note about multiple snapshots being allowed in the code. Although for now this should be fine.

I'll hold off on editing this until more input re: following up `RemoveSnapshots` in another PR or simply only supporting one value is determined.